cli/azd/pkg/auth/cache.go

```diff
@@ -16,26 +16,25 @@ type msalCacheAdapter struct {
 	cache Cache
 }
 
-func (a *msalCacheAdapter) Replace(cache cache.Unmarshaler, key string) {
-	val, err := a.cache.Read(key)
+func (a *msalCacheAdapter) Replace(ctx context.Context, cache cache.Unmarshaler, cacheHints cache.ReplaceHints) error {
+	val, err := a.cache.Read(cacheHints.PartitionKey)
 	if err != nil {
-		log.Printf("ignoring error in adapter: %v", err)
+		return err
 	}
 
 	if err := cache.Unmarshal(val); err != nil {
-		log.Printf("ignoring error in adapter: %v", err)
+		return err
 	}
+	return nil
```

Comment:
The old behavior here is that if `cache.Read` returned `err`, for example `os.ErrNotExist`, we would have logged the error, then continued on line 25 to perform `cache.Unmarshal` to overwrite the set value. So the change in #2259 *should* restore this behavior?

Unless that `cache.Unmarshal` has also changed underneath us to no longer ignore `nil` values.